vm/systemSmartContracts/delegationManager.go

```diff
@@ -312,10 +312,10 @@ func (d *delegationManager) getAllContractAddresses(args *vmcommon.ContractCallI
 
 // TODO: use all the address space
 func createNewAddress(lastAddress []byte) []byte {
-	newAddress := make([]byte, 0, len(lastAddress))
+	newAddress := make([]byte, len(lastAddress))
 	copy(newAddress, lastAddress)
 
-	for i := len(newAddress) - 1; i > 0; i++ {
+	for i := len(newAddress) - 1; i > 0; i-- {
```

Comment:
i >= 0, otherwise it won't work. Please add an unit test in which you call the function with ` lastAddr := []byte{233, 255, 255}`

Reply:
this function is going to be updated. with a version which uses all the bytes.

Reply:
no need for i >= 0 - you canot use the whole space.

Reply:
works for the scenario {0, 233, 255, 255}